### DISCUSSION TITLE
Parallel vs queued plotting

### DISCUSSION THREAD

**Discussion by berlef80** [Chia-Network/chia-blockchain#1107] [Q&A]
I am parallel plot three plots at a time.  I would like to set up queues for each of them, but trial and error has not helped me work out the way to do this, and maybe I can't do it with what the GUI functions.  It is hard for me to tell because there is no reference point in the plotting in the gui as to which plot and queued plot go together. Here is what I would like to set up

Plot 1 with Plot 1a, plot 1b, plot 1c queued to start when the previous plot ends
Plot 2 with Plot 2a, plot 2b, plot 2c queued to start when the previous plot ends 
Plot 3 with Plot 3a, plot 3b, plot 3c queued to start when the previous plot ends 

Setting plot 1 up as a queued plot is fine and works as anticipated
However. it seems when I set plot 2 up as a queued plot it just adds to the queue and does not start a new plot right then.
If I choose parallel plot then it starts, plotting but there isn't a way to queue.
In trying this out, I have had four plots start which freezes the whole machine and so I don't really want to keep experimenting and loosing plots/time along the way.

Could you explain how using the parallel and queued options I can make what I want to have happen?
Thanks!

**Comment by thatcoderguy** [comment]
Use the queue name field under advanced options when setting up plotting.
I do the following:

Setup 29 plots in serial (no parallel plotting), name the queue to "first", point it to my 500GB SSD 
Setup 29 plots in serial, name the queue to "second", point to my 1TB SSD
Setup 29 plots in serial, name the queue to "third", point to my 1TB SSD

Then when the plots reach 32%, I then create 3 more queues called "fourth", "fifth" and "sixth" - because at the end of phase 1, the temporary plots take up less space.

Each named queue runs independently, so at any one time, I have either 3 or 6 plots being created.

**Reply by pnobrega80** [reply]
Hi. So I have 1TB SSD for temporary space and 16GB RAM.  I can only plot 4 at the same time because of the temp space.
I usually make queues of X plots, like first, second, third and forth. So I get 4 plotting at the same time, same phase. 
So you're saying that I can optimize that if I start other queues after 32%? Or I totally misunderstood?

**Reply by thatcoderguy** [reply]
Correct, if you look at these charts: https://i.redd.it/w0jl1k9ml8t61.png
You can see after Phase 2, disk usage and RAM usage drops off.
I have 6 queues running in parallel on a similarly spec'd machine, you be able to squeeze 1 or 2 more queues.

**Reply by pnobrega80** [reply]
Thanks for the reply. Just started a new batch. 4 queues, and will try to start a New queue in about 4 hours (average my system is taking for phase 1). That would my system plotting 5 plots at the same time.

**Reply by MaximusChia** [reply]
Interesting . Lets be clear.
1. Queue a list of plots and put a name
2. Queue another list of plots and put name2
3. Queue another list of plots, named name3

After this, you wait until phase 1ends, and repeat process...
so you manually add this, dont use the delay option?

**Reply by pnobrega80** [reply]
That's what I understood and doing... I'm my 1TB nvme temp space I've queued 4 in parallel. When they reach the end of phase 1, I'll be adding another queue. According to the graphics after that phase the plots will require less temp space and memory...
Don't know what will happen after that. Lol.

After another plot starts in the queue...

**Reply by pnobrega80** [reply]
Well for me didn't worked well... I've started the 5th queue after the all the others passed phase 1 (around 32%).
But when the 5th queue went to 12% I was out of space. So all plots stopped...
Maybe I've done something wrong...
Deleted the 5th plot and, as usual, "Connecting to Wallet" GUI bug.  Praying for the other 4 to end

With my setup (16GB RAM and 1TB temp space) I'm only able to do 4 at the same time

**Reply by MaximusChia** [reply]
Actually i Plot 4 in parallel (2 for one SSD and set 2 for second SSD)
I have core i5 11400F =  6 cores/12 THreads
So i could run 6 in paralel in total (3x for each SSD Stick) but CPU will be full
Also 16GB Ram looks to be a limit too.
What would happen if when my 4 plottings in paralel end its first stage and then add 4? I think it Would cause bottleneck  and result in slow down. At some point you will be plotting 8 in total

**Reply by pnobrega80** [reply]
I have the same cpu. But I plot 4 in parallel with just 1tb SSD. Are your ssd 500gb? Otherwise you could plot more. 
I'm waiting for another SSD and plus 16gb so I can plot more. According to my noob calculations then I could plot only 6 because of the CPU. 

Still, it's very important, since the queues start at the same time, not to farm to the same HDD. Otherwise the final copy of the plot will waste a loooot of time because of multi copy to the hdd. My queues have different hdd destinations.

**Reply by MaximusChia** [reply]
I have 2x 1TB SSD, on each one 2 plot in paralel.
So i will try create 4 queues, 4 for each SSD. And after ending first stage will set another 4 queues. m afraid if SSD could bottleneck and slow down when i add 4 additional plots after first stage is over, 
I was planning to Plot 3 on each SSD, but RAM dont allow me actually. wonder which of 2 options is better.
And yes, copying the 4 plots to HDD tooks about 30 40 minutes, havent tested if copying one alone is faster. HDD Speed limit looks to be 125MBps, but sometimes it jump up to 220.

**Reply by pnobrega80** [reply]
With 16GB RAM and 1TB SSD  you can plot 4 at same time. Thats that I do. But more needs more RAM. My SSD keeps 90/100% most of the time, but didn't notice bottleneck. I'm taking an average of 8/9hours each plot. When copying a single file to the HDD it takes an average of 9minutes. 

Since my CPU is only 12 threads in can only do 6 same time. I know after phase 1 the plotting is made using a single thread. But that's the stuff I still didn't figured how to optimize... At least using GUI in a noobie way
